tests/postgres/alltypes_test.go

```diff
@@ -931,6 +932,43 @@ func TestTimeExpression(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestIntervalUpsert(t *testing.T) {
+	testutils.ExecuteInTxAndRollback(t, db, func(tx *sql.Tx) {
+		stmt := SELECT(Employee.AllColumns).FROM(Employee).
+			WHERE(Employee.EmployeeID.EQ(Int(1)))
+
+		//Validate initial dataset
+		var windy model.Employee
+		err := stmt.Query(db, &windy)
+		assert.Equal(t, windy.EmployeeID, int32(1))
+		assert.Equal(t, windy.FirstName, "Windy")
+		assert.Equal(t, windy.LastName, "Hays")
+		assert.Equal(t, *windy.PtoAccrual, "22:00:00")
+		assert.Nil(t, err)
+		windy.PtoAccrual = ptr.Of("3h")
+		//Update data
+		updateStmt := Employee.UPDATE(Employee.PtoAccrual).SET(
+			Employee.PtoAccrual.SET(INTERVAL(3, HOUR)),
+		).WHERE(Employee.EmployeeID.EQ(Int(1))).RETURNING(Employee.AllColumns)
+
+		err = updateStmt.Query(db, &windy)
+		err = stmt.Query(db, &windy)
+		assert.Nil(t, err)
+		assert.Equal(t, *windy.PtoAccrual, "03:00:00")
+		//Upsert dataset with a different value
+		windy.PtoAccrual = ptr.Of("22h")
+		insertStmt := Employee.INSERT(Employee.AllColumns).
+			MODEL(&windy).
+			ON_CONFLICT(Employee.EmployeeID).
+			DO_UPDATE(SET(
+				Employee.PtoAccrual.SET(Employee.EXCLUDED.PtoAccrual),
+			)).RETURNING(Employee.AllColumns)
+		err = insertStmt.Query(db, &windy)
+		assert.Nil(t, err)
+		assert.Equal(t, *windy.PtoAccrual, "22:00:00")
+	})
+}
+
```

Comment:
There might be an overall bug on how ExecuteInTrxAndRollback works, I had to make the final test use the original value so it won't affect other tests.  It doesn't seem to rollback the data.

Reply:
You need to pass `tx` to Query method for rollback to work:
```golang
insertStmt.Query(tx, &windy)
```

Reply:
Doh.  Thanks.